In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

In [2]:
from google.colab import files
uploaded = files.upload()

Saving USA_Housing.csv to USA_Housing.csv


In [4]:
import io
df= pd.read_csv(io.BytesIO(uploaded['USA_Housing.csv']))
# Dataset is now stored in a Pandas Dataframe

In [5]:
df

Avg. Area Income  Avg. Area House Age  Avg. Area Number of Rooms  \
0          79545.45857             5.682861                   7.009188   
1          79248.64245             6.002900                   6.730821   
2          61287.06718             5.865890                   8.512727   
3          63345.24005             7.188236                   5.586729   
4          59982.19723             5.040555                   7.839388   
...                ...                  ...                        ...   
4995       60567.94414             7.830362                   6.137356   
4996       78491.27543             6.999135                   6.576763   
4997       63390.68689             7.250591                   4.805081   
4998       68001.33124             5.534388                   7.130144   
4999       65510.58180             5.992305                   6.792336   

      Avg. Area Number of Bedrooms  Area Population         Price  \
0                             4.09      23086.80050  1.059034e+06   
1                             3.09      40173.07217  1.505891e+06   
2                             5.13      36882.15940  1.058988e+06   
3                             3.26      34310.24283  1.260617e+06   
4                             4.23      26354.10947  6.309435e+05   
...                            ...              ...           ...   
4995                          3.46      22837.36103  1.060194e+06   
4996                          4.02      25616.11549  1.482618e+06   
4997                          2.13      33266.14549  1.030730e+06   
4998                          5.44      42625.62016  1.198657e+06   
4999                          4.07      46501.28380  1.298950e+06   

                                                Address  
0     208 Michael Ferry Apt. 674\nLaurabury, NE 3701...  
1     188 Johnson Views Suite 079\nLake Kathleen, CA...  
2     9127 Elizabeth Stravenue\nDanieltown, WI 06482...  
3                             USS Barnett\nFPO AP 44820  
4                            USNS Raymond\nFPO AE 09386  
...                                                 ...  
4995                   USNS Williams\nFPO AP 30153-7653  
4996              PSC 9258, Box 8489\nAPO AA 42991-3352  
4997  4215 Tracy Garden Suite 076\nJoshualand, VA 01...  
4998                          USS Wallace\nFPO AE 73316  
4999  37778 George Ridges Apt. 509\nEast Holly, NV 2...  

[5000 rows x 7 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Avg. Area Income              5000 non-null   float64
 1   Avg. Area House Age           5000 non-null   float64
 2   Avg. Area Number of Rooms     5000 non-null   float64
 3   Avg. Area Number of Bedrooms  5000 non-null   float64
 4   Area Population               5000 non-null   float64
 5   Price                         5000 non-null   float64
 6   Address                       5000 non-null   object 
dtypes: float64(6), object(1)
memory usage: 273.6+ KB


In [7]:
# what is the % of missing values in the dataset?
def percent_missing(df):

    # Calculate total number of cells in dataframe
    totalCells = np.product(df.shape)

    # Count number of missing values per column
    missingCount = df.isnull().sum()

    # Calculate total number of missing values
    totalMissing = missingCount.sum()

    # Calculate percentage of missing values
    print("The dataset contains", round(((totalMissing/totalCells) * 100), 2), "%", "missing values.")

percent_missing(df)

The dataset contains 0.0 % missing values.


In [8]:
df.isna().sum()

Avg. Area Income                0
Avg. Area House Age             0
Avg. Area Number of Rooms       0
Avg. Area Number of Bedrooms    0
Area Population                 0
Price                           0
Address                         0
dtype: int64

No missing values

**Question 3**

**MLR** <br>
Here : Y = b0 + b1 * x1 + b2 * x2 + b3 * x3 + …… bn * xn 
Y = Dependent variable and x1, x2, x3, …… xn = multiple independent variables 

In [9]:
from sklearn.model_selection import train_test_split
from sklearn import datasets
from sklearn import svm

In [14]:
df2 = df.drop("Address",  axis=1)

In [15]:
# Create X (all the feature columns)
X = df2.drop("Price",  axis=1)

# Create y (the target column)
y = df2["Price"]

In [17]:
from sklearn import linear_model
import statsmodels.api as sm
regr = linear_model.LinearRegression()
regr.fit(X, y)

LinearRegression()

In [19]:
print('Intercept: \n', regr.intercept_)
print('Coefficients: \n', regr.coef_)

# with statsmodels
X = sm.add_constant(X) # adding a constant
 
model = sm.OLS(y, X).fit()
predictions = model.predict(X) 
 
print_model = model.summary()
print(print_model)

Intercept: 
 -2637299.033331454
Coefficients: 
 [2.15780494e+01 1.65637027e+05 1.20659949e+05 1.65113905e+03
 1.52007439e+01]
                            OLS Regression Results                            
Dep. Variable:                  Price   R-squared:                       0.918
Model:                            OLS   Adj. R-squared:                  0.918
Method:                 Least Squares   F-statistic:                 1.119e+04
Date:                Tue, 20 Sep 2022   Prob (F-statistic):               0.00
Time:                        15:35:16   Log-Likelihood:                -64714.
No. Observations:                5000   AIC:                         1.294e+05
Df Residuals:                    4994   BIC:                         1.295e+05
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                   coef    std err          t      P>|t|      [0.025

`Price = (-2637299.03) + (2.15780494e+01)*Avg. Area Income   + (1.65637027e+05)*Avg. Area House Age   + (1.20659949e+05)*Avg. Area Number of Rooms + (1.65113905e+03)*Avg. Area Number of Bedrooms + (1.52007439e+01)*Area Population `